lib/miq_expression.rb

```diff
@@ -598,43 +606,38 @@ def self.value2human(val, options = {})
     ret
   end
 
-  def self.operands2rubyvalue(operator, ops, context_type)
-    # puts "Enter: operands2rubyvalue: operator: #{operator}, ops: #{ops.inspect}"
+  def self.quote_by(operator, value, column_type = nil)
+    if UNQUOTABLE_OPERATORS.map(&:downcase).include?(operator)
+      value
+    else
+      quote(value, column_type.to_s)
+    end
+  end
 
+  def self.operands2rubyvalue(operator, ops, context_type)
     if ops["field"]
       if ops["field"] == "<count>"
         ["<count>", quote(ops["value"], "integer")]
       else
         target = parse_field_or_tag(ops["field"])
         col_type = target&.column_type || "string"
-        if context_type == "hash"
-          val = ops["field"].split(".").last.split("-").join(".")
-          fld = "<value type=#{col_type}>#{val}</value>"
-        else
-          fld = "<value ref=#{target.model.to_s.downcase}, type=#{col_type}>#{target.tag_path_with}</value>"
-        end
-        if ["like", "not like", "starts with", "ends with", "includes", "regular expression matches", "regular expression does not match"].include?(operator)
-          [fld, ops["value"]]
-        else
-          [fld, quote(ops["value"], col_type.to_s)]
-        end
+
+        [if context_type == "hash"
+           "<value type=#{col_type}>#{ops["field"].split(".").last.split("-").join(".")}</value>"
+         else
+           "<value ref=#{target.model.to_s.downcase}, type=#{col_type}>#{target.tag_path_with}</value>"
+         end, quote_by(operator, ops["value"], col_type)]
```

Comment:
lol. I was thinking about duplicating the `quote_by` line, but this is good enough